### DISCUSSION TITLE
[Feature] Primary album (reflected in storage template)

### DISCUSSION THREAD

**Discussion by caeci11ius** [immich-app/immich#10839] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I would like to be able to specify a "primary album" for a given album that would then be reflected in the storage template for all contained photos. 

What I'm hoping to achieve is the ability to import a bunch of images, add them to albums based on events they were part of, mark those albums as primary and then have those live on the storage in some kind of intelligible manner (such that if I ever stopped using immich in future, I could understand my library structure more readily).

If the existing album tag in the storage template used the first album added instead of the last, that would be something of a workaround...but ideally this would enable moving things around via the webui too by simply ticking a "primary" box on an album or similar (which would then have all contained images move there on storage). As for using the last album to drive that feature, this breaks the ability to add photos to multiple albums without having them move around on storage...

I should also add that I don't just want to use a read only existing library on disk. I do have a somewhat organised library from in the past, but since then years of switching through iPhoto and Google photos have kind-of made a mess of things from certain periods of my life. I'm hoping to use immich's dupe detection features to tidy this up - but I understand the important of a clear file structure to the ability to move on in future if needed!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Szeraax** [comment]
To do this, we'd probably want a photo level "primary album" foreign key. And when there are multiple albums that an image belongs to, identify which album is the primary one via a floating symbol or even some text like "Primary"
![image](https://github.com/user-attachments/assets/653e674a-1050-454a-9a2d-358c2edfcbef)

Further, we could use the existing album select tool and add in the capability to make this the primary album for the select assets like so (obviously with a better icon):

![image](https://github.com/user-attachments/assets/b2c80abd-7668-432a-8350-a44d3d166054)

What do you think of that approach?

**Reply by Szeraax** [reply]
Then obviously the storage template tool would ALSO have to check what album is the primary one instead of checking to see which album was created most recently. And by default the newest created album is the one that photos get their primary set to.

**Reply by caeci11ius** [reply]
That looks great. Maybe even have a button in the album itself to just make that the primary album for all content. So you could easily make an album the primary for all images contained, or select some images and set a primary album just for those. But yes, that's pretty well the idea :-).

**Comment by Szeraax** [comment]
My current work around is adding pgAdmin to my docker compose and manually editing the album createdAt property in the albums to prioritize up/down.

**Comment by radh21301** [comment]
Same as:

https://github.com/immich-app/immich/discussions/6810

**Comment by caeci11ius** [comment]
Not really the same, unless you account for nthamling's comment a few down from the top (at which point you could say they're similar). Certainly the original request doesn't request primary albums, though this request would probably solve for their use-case too...

**Reply by radh21301** [reply]
You are right. But yeah, at its core, both are asking for the same thing. Some way to control which album is used in the storage template.
It is something I need badly. Right now my images keep moving around if I add them to another album (they end up in the latest album that it was added to)

**Comment by ChilledSlim** [comment]
It makes little sense to have a file keep moving around by using most recently created album. It should either be the first album (which is usually the event it was associated with) or allow a customizable primary album.

2023-05 jacks birthday
2025-01 concert of bandname with Jack
2025-10 Jack's wedding reel
2025-12 Christmas photo frame reel 2025

Obviously the first two are the primary albums. These are the oldest albums that were created when the photos were first uploaded.  This is where they make sense to be.

Then you can put them into an album of photos of Jack for a collage (wedding reel) or the Christmas photo frame.  The primary shouldn’t change.  It should NEVER change unless explicitly told to change it.  We shouldn't be pulling these out of jack's birthday and concert with jack when we make the wedding reel.  Yes these could be tags arguably.

OLDEST makes sense.  NEWEST makes little sense.

**Reply by Szeraax** [reply]
This is a very good point. "Jack's Wedding Reel" shouldn't pull all the photos from his life out of their original folders and into this brand new one on storage. 

However, it is a LOT easier if you DO want to move photos to just create a new album and associate them vs having to delete ALL albums that are older than X album that you want as primary. From that side, I can understand why Immich would NEVER want to make the oldest album the one that is used by storage template. 

In my mind, it will either stay as is, or they need to add a way to select which album will be the stored location. If they did add the ability to select primary album, then I think doing the oldest album by default makes a LOT more sense than the newest one.

**Reply by ChilledSlim** [reply]
Agreed.  OLDEST makes sense.  NEWEST does not.  But being closed, I doubt this issue will see traction.
